dags/iop/parser.py

```diff
@@ -46,6 +47,16 @@ def __init__(self) -> None:
                 attribute="count",
                 extra_function=lambda x: [parse_to_int(x)] if parse_to_int(x) else None,
             ),
+            CustomExtractor(
+                destination="date_published",
+                extraction_function=self._get_date_published,
+                default_value=datetime.date.today().isoformat(),
```

Comment:
you're putting today's date in case of missing any date element. Let's say the day is missing, but you have a year 2021 and a month 7. Then you are putting the current date as a published date, which is totally different than the real one. Are you sure it is what you want?

Reply:
mmm... I am not sure, actually. It makes more sense to return at least a part of the date. But then, if there is just a day, or there are just a day and month, what's then? In that case, is better to return today's date, no?

Reply:
No, I'll apply the same logic as on inspire: https://github.com/inspirehep/inspire-utils/blob/2ae4722ce5b65c3d168b0870c2c19e03d0144954/inspire_utils/date.py#L38
So:
* year is mandatory,
* month is mandatory if there's a day.
And for the default date, I think you should ask Anne, but IMO it's a bad idea to set today's date.

Reply:
Talked with Anne:
1) if we don't have a day -> we leave just a year and a month
2) if we don't have a month -> we leave just a year
3) if we don't have a year -> it stops. Year from date_published is is the value of  journal_year which is required